share/spack/setup-env.sh

```diff
@@ -202,9 +202,10 @@ export SPACK_ROOT=${_sp_prefix}
 # Determine which shell is being used
 #
 function _spack_determine_shell() {
-	ps -p $$ | tail -n 1 | awk '{print $4}' | sed 's/^-//' | xargs basename
+        ps -p $$ | tail -n 1 | awk '{print $4}' | sed 's/^-//' | xargs basename
 }
-export SPACK_SHELL=$(_spack_determine_shell)
+: ${SPACK_SHELL:=$(_spack_determine_shell)}
```

Comment:
My Bash is a little rusty. What purpose does the leading colon serve here?

Reply:
We should also add a comment saying that it can be overridden by setting this env var in your own .bashrc

Reply:
The one at the start or by the `=`? The `:` at the beginning seems so subtle that I can't say I really understand it, but since `:` is essentially a noop or merely `true`, it kind of formally turns the variable substitution into a command. I think there are examples where using `${SOMETHING='value'}` can somehow try to execute `value` unless you prepend it with `:`. It seems to work without it for me, but the colon seems to be the convention there. The `:` before the equal allows `SPACK_SHELL` to be set whether it is unset or equal to the empty string, where no colon there would only set it if it is unset at that moment.

Reply:
Ah, I see it. Try this:
```
#!/bin/bash
${SOMETHING:=hello}
```
and
```
#!/bin/bash
: ${SOMETHING:=hello}
```

Reply:
I hate that syntax! More readable would be:
```
SOMETHING=${SOMETHING:-hello}
```

Reply:
I can't say I agree, but I have no problem changing it. Bash syntax is all arbitrary to me. I'm guessing people will hate this syntax as well.